### DISCUSSION TITLE
how to link msquic in windows

### DISCUSSION THREAD

**Discussion by StargazerWayne** [microsoft/msquic#4048] [Q&A]
I have been using msquic as a submodule to enable quic feature. I simply use cmake cmd `add_subdirectory` and `target_link_libraries` to use quic lib in linux, they works fine.

When it comes to windows, the msvc compiler keeps telling that it can't find some functions and variables in WinSDK. (I am using vscode with powershell)
But I can compiler msquic separately and run the quic sample which indicates WinSDK is installed and is available for cmake and msvc. 

I know I may compiler msquic separately and use `find_library` to use the lib, but I just want to keep consistent with what I do on Linux. What should I do with cmake to link msquic in windows, (I know it sounds like a cmake thing, but :/ ) any hint will be appreciated. Thank you!

**Comment by nibanks** [comment]
Please take a look at https://github.com/microsoft/quicreach or https://github.com/nibanks/msh3 for good examples. They both submodule MsQuic and build cross-platform. Let me know if you have any other questions after looking at them.

**Reply by StargazerWayne** [reply]
Thank you for your advices. I have tried a several times to imitate how the examples use msquic, but I still didn't make it. 

The compile error are undefine of thread like and crypt like function and macros like `GetCurrentProcessorNumberEx`, `SetThreadGroupAffinity`, `GetThreadGroupAffinity`, `BCRYPT_HMAC_SHA512_ALG_HANDLE`, `BCRYPT_SHA256_ALG_HANDLE`. Not all windows sdk is unavailable. Why is that？ 

What might the project that submodule msquic do that will affect that. I have spent quite a time, but still can't figure out why? Do you have any clue?

**Reply by nibanks** [reply]
You need to use a fairly recent (WS2019) SDK to build. What are you using?

**Reply by StargazerWayne** [reply]
I use the official winsdksetup.exe to install sdk , I believe it's the latest. ![image](https://github.com/microsoft/msquic/assets/53253973/8e26835e-51ef-4dc6-b7f1-7b7972f8a2a4)
I can find the 'missing' functions in local but not the MSVC compiler.
![image](https://github.com/microsoft/msquic/assets/53253973/4e07d304-975f-4a4e-b1b8-dab24d6f8222)
